test/explain/algorithm/test_gnn_explainer.py

```diff
@@ -0,0 +1,389 @@
+        # pass arguments in the right order
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        return self.lin(x)
+
+
+class GNN_classification_singleoutput(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 7)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right order
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        x = self.lin(x)
+        return x.log_softmax(dim=1)
+
+
+class GNN_classification_multioutput(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 7)
+        self.lin2 = Linear(16, 7)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right order
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        out_1 = self.lin(x).log_softmax(dim=1)
+        out_2 = self.lin2(x).log_softmax(dim=1)
+        return torch.stack([out_1, out_2], dim=0)
+
+
+# --------------------------------------------------------------------
+# check explanations
+# --------------------------------------------------------------------
+def check_explanation(
+    edge_mask_type,
+    node_mask_type,
+    x,
+    edge_index,
+    explanation,
+):
+    if node_mask_type == MaskType.attributes:
+        assert explanation.node_features_mask.shape == x.shape
+        assert explanation.node_features_mask.min() >= 0
+        assert explanation.node_features_mask.max() <= 1
+    elif node_mask_type == MaskType.object:
+        assert explanation.node_mask.shape == x.shape[0]
+        assert explanation.node_mask.min() >= 0
+        assert explanation.node_mask.max() <= 1
+    elif node_mask_type == MaskType.common_attributes:
+        assert explanation.node_features_mask.shape == x.shape
+        assert explanation.node_features_mask.min() >= 0
+        assert explanation.node_features_mask.max() <= 1
+        assert explanation.node_features_mask[
+            0] == explanation.node_features_mask[1]
+
+    if edge_mask_type == MaskType.object:
+        assert explanation.edge_mask.shape == (edge_index.shape[1], )
+        assert explanation.edge_mask.min() >= 0
+        assert explanation.edge_mask.max() <= 1
+
+
+# type of masks allowed for GNNExplainer
+node_mask_types = ["attributes", "object", "common_attributes"]
+edge_mask_types = ["object", None]
+return_types_classification = ["log_probs", "raw", "probs"]
```

Comment:
I think you forgot to use this

Reply:
Sorry, this refers to `return_types_classification`

Reply:
good catch, forgot to switch for classification 👍🏽